### PR TITLE
Store data

### PR DESCRIPTION
**Pull request by bryanvanwijk**

### CODE REVIEW

File: android/app/src/main/java/com/gpslogger/LocationModule.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+public class LocationModule extends ReactContextBaseJavaModule {
+
+
+    public LocationModule(ReactApplicationContext reactContext) {
+        super(reactContext);
+    }
+
+    @Override
+    public String getName() {
+        return "Location";
```
Comment (jochem725) [inline android/app/src/main/java/com/gpslogger/LocationModule.java]: Maybe we can call this NativeLocation, to be more specific.

File: android/app/src/main/java/com/gpslogger/LocationModule.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    @ReactMethod
+    public void getGPSLocation(final Callback position) {
+
+        String context = Context.LOCATION_SERVICE;
+        LocationManager locationManager = (LocationManager) MainApplication.getAppContext().getSystemService(context);
+
+        locationManager.requestSingleUpdate(LocationManager.GPS_PROVIDER, new LocationListener(){
+
+            public void onLocationChanged(Location location) {
+                JSONObject json = new JSONObject();
```
Comment (jochem725) [inline android/app/src/main/java/com/gpslogger/LocationModule.java]: Maybe the creation of the JSON output can be done in a separate method.

File: js/LocationSampler.ts [hunk 3]
```diff
@@ -40,6 +44,16 @@ export default class LocationSampler {
             this.timerId = -1;
             this.running = false;
         }
+        console.log(RNFS.ExternalDirectoryPath);
+        var path = RNFS.ExternalDirectoryPath + '/' + this.measurementName + '.json';
+
+        var data = JSON.stringify({accuracy: 1, battery_start: 99, battery_end: 67, samples: this.samples});
```
Comment (jochem725) [inline js/LocationSampler.ts]: Since these default values have no use case yet, maybe we can omit them for now.

File: js/LocationSampler.ts [hunk 4]
```diff
@@ -53,10 +67,16 @@ export default class LocationSampler {
      * Adds a geolocation sample to the list of collected samples.
      */
     private getGeoLocation(): void {
-        navigator.geolocation.getCurrentPosition(
-            (position) => {console.log(position.coords.latitude); this.samples.push(position)},
-            (error) => {console.log(error);},
-            {enableHighAccuracy: this.highAccuracy}
+        NativeModules.Location.getGPSLocation(
+            (err, position) => {
+                if(!err) {
```
Comment (jochem725) [inline js/LocationSampler.ts]: `if (!err)`

**Review by jochem725 — CHANGES_REQUESTED** [review]

**Review by jochem725 — DISMISSED** [review]

**Review by doriandekoning — APPROVED** [review]

**Review by jochem725 — APPROVED** [review]

**Pull request merged**